quarkus/runtime/src/test/java/org/keycloak/provider/quarkus/ConfigurationTest.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.provider.quarkus;
+
+import static org.junit.Assert.assertEquals;
+
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import io.smallrye.config.SmallRyeConfig;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+import org.junit.After;
+import org.junit.Test;
+import org.keycloak.Config;
+import org.keycloak.configuration.KeycloakConfigSourceProvider;
+import org.keycloak.configuration.MicroProfileConfigProvider;
+
+import io.quarkus.runtime.configuration.ConfigUtils;
+import io.smallrye.config.SmallRyeConfigProviderResolver;
+
+public class ConfigurationTest {
+
+    private static final Properties SYSTEM_PROPERTIES = (Properties) System.getProperties().clone();
+    private static final Map<String, String> ENVIRONMENT_VARIABLES = new HashMap<>(System.getenv());
+
+    @SuppressWarnings("unchecked")
+    public static void putEnvVar(String name, String value) {
+        Map<String, String> env = System.getenv();
+        Field field = null;
+        try {
+            field = env.getClass().getDeclaredField("m");
```

Comment:
Will this work for various java versions etc? I found this https://stackoverflow.com/questions/8168884/how-to-test-code-dependent-on-environment-variables-using-junit and I wonder if its better alternative?

Reply:
On the other hand, I guess we may not want to introduce dependency on another library even for tests? Maybe just inspire from it or just go with you what you have (if you know it will work in various java versions we need to support).

Reply:
Yeah, for now it works fine. I did not want to introduce another lib.

Reply:
Ok, that works for me. Thanks